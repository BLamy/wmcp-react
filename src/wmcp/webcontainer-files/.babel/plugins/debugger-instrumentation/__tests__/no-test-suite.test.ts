import { describe, it, expect } from 'vitest';

import { runSingle  } from './helpers/runSingle';

describe('debuggerInstrumentation ‑ single file', () => {
//   it('injects runtime stub & logs a basic step', () => {
//     const steps = runSingle('const a = 1;', 'basic_stub.ts');
//     expect(globalThis.__recordStep).toBeTypeOf('function');
//     expect(globalThis.__resetStepCounter).toBeTypeOf('function');
//     expect(steps).toMatchSnapshot();
//     expect(steps).toHaveLength(1);
//     expect(steps[0].vars).toMatchObject({ a: 1 });
//     expect(steps[0].line).toBe(1);
//     expect(steps[0].stepNumber).toBe(1);
//   });

  it('honours maxVars', () => {
    const [step] = runSingle('const a=1,b=2,c=3,d=4;', 'maxVars.ts', { maxVars: 2 });
    expect(Object.keys(step.vars).length).toBeLessThanOrEqual(2);
  });

  it('skips classic‑function arguments', () => {
    const steps = runSingle(`
      function foo(x, y){ return x + y }
      foo(4,5)
    `);
    expect(steps.some(s => 'arguments' in s.vars)).toBe(false);
  });

  it('instruments arrow fn implicit return', () => {
    const steps = runSingle(`
      const add = (m,n) => m + n;
      add(2,3);
    `);
    const got = steps.find(s => 'm' in s.vars && 'n' in s.vars);
    expect(got?.vars).toMatchObject({ m: 2, n: 3 });
  });

  it('wraps single‑line if‑statement', () => {
    const steps = runSingle(`
      let x = 0; if (x===0) x = 1;
    `);
    expect(steps.length).toBeGreaterThanOrEqual(2);
  });

  // it('sanitises funky suite / test names', () => {
  //   const steps = runSingle(
  //     `describe('My Suite/Weird:Name.v1', () => {
  //       test('weird test name: 1/2', () => { 
  //         const v = 42
  //       });
  //     });
  //     `,
  //     'snippet.js',
  //     { suiteName: 'My Suite/Weird:Name.v1' },
  //   );
  //   const hit = steps.find(s => s.test === 'weird test name: 1/2');
  //   expect(hit).toBeTruthy();
  // });

  it('increments stepNumber sequentially', () => {
    const steps = runSingle(`
      let sum=0; for(let i=0;i<5;i++) sum+=i;
    `);
    const nums = steps.map(s => s.stepNumber);
    expect(nums).toEqual([...nums].sort((a,b)=>a-b));
  });

  it('multi‑line if‑block detail (snapshot)', () => {
    const steps = runSingle(`
      let x = 0;
      if (x === 0) {
        x = 1;
      } else if (x === 1) {
        x = 2;
      } else { x = 3; }
    `);
    expect(steps).toMatchSnapshot();
  });

  it('captures loop‑scoped vars each iteration', () => {
    const steps = runSingle(`
      for(let j=0;j<3;j++){ const squared=j*j }
    `);
    const loopSteps = steps.filter(s => 'j' in s.vars && 'squared' in s.vars);
    expect(loopSteps).toHaveLength(3);
  });

  it('handles destructuring without TDZ', () => {
    const steps = runSingle(`const { a, b: renamed } = { a: 7, b: 9 };`);
    const step = steps.find((s) => 'a' in s.vars && 'renamed' in s.vars);
    expect(step?.vars).toMatchObject({ a: 7, renamed: 9 });
  });
});
