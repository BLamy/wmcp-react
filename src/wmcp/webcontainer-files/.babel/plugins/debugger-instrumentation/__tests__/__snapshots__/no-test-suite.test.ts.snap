// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`debuggerInstrumentation ‑ single file > multi‑line if‑block detail (snapshot) 1`] = `
[
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.js",
    "line": 2,
    "sourceLine": "const { describe, test } ='vitest';",
    "stepNumber": 1,
    "suite": "My_Suite_Weird_Name_v1",
    "test": "UnknownTest",
    "vars": {},
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 2,
    "sourceLine": "interface Foo{bar:string}",
    "stepNumber": 1,
    "suite": "RootSuite",
    "test": "UnknownTest",
    "vars": {},
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 2,
    "sourceLine": "let x = 0;",
    "stepNumber": 1,
    "suite": "single",
    "test": "UnknownTest",
    "vars": {
      "x": 0,
    },
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 3,
    "sourceLine": "function baz(f:Foo){",
    "stepNumber": 2,
    "suite": "RootSuite",
    "test": "UnknownTest",
    "vars": {
      "f": {
        "bar": "hi",
      },
    },
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 3,
    "sourceLine": "if (x === 0) {",
    "stepNumber": 2,
    "suite": "single",
    "test": "UnknownTest",
    "vars": {
      "x": 0,
    },
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 4,
    "sourceLine": "const x = f.bar;",
    "stepNumber": 3,
    "suite": "RootSuite",
    "test": "UnknownTest",
    "vars": {
      "f": {
        "bar": "hi",
      },
      "x": "hi",
    },
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 4,
    "sourceLine": "x = 1;",
    "stepNumber": 3,
    "suite": "single",
    "test": "UnknownTest",
    "vars": {
      "x": 1,
    },
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 7,
    "sourceLine": "baz({bar:'hi'});",
    "stepNumber": 4,
    "suite": "RootSuite",
    "test": "UnknownTest",
    "vars": {},
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 2,
    "sourceLine": "let sum=0; for(let i=0;i<5;i++) sum+=i;",
    "stepNumber": 4,
    "suite": "single",
    "test": "UnknownTest",
    "vars": {
      "i": 1,
      "sum": 1,
    },
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 2,
    "sourceLine": "let sum=0; for(let i=0;i<5;i++) sum+=i;",
    "stepNumber": 5,
    "suite": "single",
    "test": "UnknownTest",
    "vars": {
      "i": 1,
      "sum": 1,
    },
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 2,
    "sourceLine": "let sum=0; for(let i=0;i<5;i++) sum+=i;",
    "stepNumber": 6,
    "suite": "single",
    "test": "UnknownTest",
    "vars": {
      "i": 2,
      "sum": 3,
    },
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 2,
    "sourceLine": "let sum=0; for(let i=0;i<5;i++) sum+=i;",
    "stepNumber": 7,
    "suite": "single",
    "test": "UnknownTest",
    "vars": {
      "i": 2,
      "sum": 3,
    },
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 2,
    "sourceLine": "let sum=0; for(let i=0;i<5;i++) sum+=i;",
    "stepNumber": 8,
    "suite": "single",
    "test": "UnknownTest",
    "vars": {
      "i": 3,
      "sum": 6,
    },
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 2,
    "sourceLine": "let sum=0; for(let i=0;i<5;i++) sum+=i;",
    "stepNumber": 9,
    "suite": "single",
    "test": "UnknownTest",
    "vars": {
      "i": 3,
      "sum": 6,
    },
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 2,
    "sourceLine": "let sum=0; for(let i=0;i<5;i++) sum+=i;",
    "stepNumber": 10,
    "suite": "single",
    "test": "UnknownTest",
    "vars": {
      "i": 4,
      "sum": 10,
    },
  },
  {
    "file": "/Users/brettlamy/Dev/wmcp-react/snippet.ts",
    "line": 2,
    "sourceLine": "let sum=0; for(let i=0;i<5;i++) sum+=i;",
    "stepNumber": 11,
    "suite": "single",
    "test": "UnknownTest",
    "vars": {
      "i": 4,
      "sum": 10,
    },
  },
]
`;
